prov/efa/src/rxr/rxr_domain.c

```diff
@@ -197,7 +200,9 @@ int rxr_mr_regattr(struct fid *domain_fid, const struct fi_mr_attr *attr,
 	}
 
 	/* Call shm provider to register memory */
-	if (rxr_env.enable_shm_transfer && !key_exists) {
+	if (rxr_env.enable_shm_transfer
+	    && !key_exists
+	    && attr->iface == FI_HMEM_SYSTEM) {
```

Comment:
I think there need to be a correspond change in rxr_mr_close() that does not close shm mr for CUDA?

Reply:
I thought I had that commit in this PR too, but got missed when consolidating my PRs. Will add it.